rn/Teacher/ios/Submissions/PostSettingsViewController.swift

```diff
@@ -0,0 +1,137 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2019-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import UIKit
+import Core
+
+class PostSettingsViewController: UIViewController {
+    @IBOutlet weak var menu: HorizontalMenuView!
+    @IBOutlet weak var scrollView: UIScrollView!
+    @IBOutlet weak var menuHeight: NSLayoutConstraint!
+    private var postGradesViewController: PostGradesViewController!
+    private var hideGradesViewController: HideGradesViewController!
+    @IBOutlet weak var containerA: UIView!
+    @IBOutlet weak var containerB: UIView!
+
+    enum MenuItem: Int {
+        case post, hide
+    }
+
+    static func create() -> PostSettingsViewController {
+        let controller = loadFromStoryboard()
+        return controller
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = NSLocalizedString("Post Settings", comment: "")
+        configureMenu()
+        configurePost()
+        configureHide()
+        scrollView.delegate = self
+    }
+
+    func configureMenu() {
+        menu.delegate = self
+        menuHeight.constant = 1.0 / UIScreen.main.scale
+    }
+
+    func configurePost() {
+        postGradesViewController = PostGradesViewController.create()
+        embed(postGradesViewController, in: containerA)
+    }
+
+    func configureHide() {
+        hideGradesViewController = HideGradesViewController.create()
+        embed(hideGradesViewController, in: containerB)
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        let ratio = self.scrollView.contentOffsetRatio
+        coordinator.animate(alongsideTransition: { [weak self] _ in
+            ratio.x >= 0.5 ? self?.showHide() : self?.showPost()
+            self?.menu.reload()
+            }, completion: nil)
```

Comment:
nitpick: indentation (we need a linter for this)

Reply:
this is how xcode is formatting it.  When i format it differently and then format w/ the IDE, it changes to Xcode's format.  I agree that a linter would be great.  I'd rather that be outside the scope of this pr.  I would not propose we fight the formatting of the IDE as that seems a never ending battle.